splunk-otel-android/src/main/java/com/splunk/rum/StartTypeAwareSpanStorage.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import static com.splunk.rum.SplunkRum.LOG_TAG;
+
+import android.util.Log;
+import io.opentelemetry.android.instrumentation.activity.VisibleScreenTracker;
+import java.io.File;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/***
+ * Store span files to /span/background/@uniqueId/ for spans created when the app started in the background.
+ * If the app is brought to foreground and the same session ID still in use, the background spans are moved to /span for eventual sending.
+ * If the app still in the background until process-kill, the background span files will eventually be deleted by the @DeviceSpanStorageLimiter.
+ */
+public class StartTypeAwareSpanStorage implements SpanStorage {
+
+    private final VisibleScreenTracker visibleScreenTracker;
+    private final FileUtils fileUtils;
+    private final String uniqueId;
+    private final File rootDir;
+    private final File spanDir;
+
+    public StartTypeAwareSpanStorage(
+            VisibleScreenTracker visibleScreenTracker, FileUtils fileUtils, File rootDir) {
+        this.visibleScreenTracker = visibleScreenTracker;
+        this.fileUtils = fileUtils;
+        this.rootDir = rootDir;
+        this.spanDir = fileUtils.getSpansDirectory(rootDir);
+        this.uniqueId = UUID.randomUUID().toString();
+        // if new id then background span directory with old ids can be deleted
+        cleanUpUnsentBackgroundSpan();
```

Comment:
I think it's usually considered bad practice to have a side effect like this just from instance creation. I can think of two options -- 

1. Make the method accessible and call it from the `RumInitializer` right after creation.
2. Make a static `create()` factory method that does the two things -- create the instance to return and also, before returning, calls the cleanup.

I think 2 is slightly cleaner.

Reply:
I agree, 2nd solution is better